src/lib/menu/menu-directive.ts

```diff
@@ -139,7 +139,7 @@ export class MdMenu implements AfterContentInit, MdMenuPanel, OnDestroy {
   }
 
   /** Event emitted when the menu is closed. */
-  @Output() close = new EventEmitter<void | 'click' | 'keydown'>();
+  @Output() close = new EventEmitter<void | 'click' | 'destroy' | 'keydown'>();
```

Comment:
The idea with this one was to emit the reason only if it's user-generated. Given that there is no special behavior for `destroy`, can you revert it back to `void | 'click' | 'keydown'`?

Reply:
Done. I was also wondering if it's a bug that destroy will emit a close event when the menu is already closed?

Reply:
It's there to notify any child menus to close.

Reply:
But child menus won't be open unless the menu itself is open? Anyway, I'm more than happy to leave it as it is, as it doesn't really affect any use cases that I can think of.

Reply:
In most cases the children won't be open without a parent, but it's possible for the parent to be destroyed while the children are still open.